src/platform/Tizen/PosixConfig.cpp

```diff
@@ -101,7 +101,14 @@ CHIP_ERROR PosixConfig::ReadConfigValue(Key key, uint64_t & val)
 CHIP_ERROR PosixConfig::ReadConfigValueStr(Key key, char * buf, size_t bufSize, size_t & outLen)
 {
     VerifyOrReturnError(buf != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    return PersistedStorage::KeyValueStoreMgr().Get(key.Name, buf, bufSize, &outLen);
+
+    auto err = PersistedStorage::KeyValueStoreMgr().Get(key.Name, buf, bufSize, &outLen);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+    VerifyOrReturnError(outLen > 0, CHIP_ERROR_PERSISTED_STORAGE_FAILED);
+    outLen--; // Account for null terminator
```

Comment:
Wouldn't it be better to not store the null-terminator (so you don't have to strndup in WriteConfigValueStr), but then here ensure that bufSize has one extra byte for the null-terminator and add it in this code?

Reply:
I knew that I'm missing something here... :D of course it would be better

Reply:
Updated. Many thanks for review! :)